packages/gatsby-theme-docs/src/templates/release-notes-list.js

```diff
@@ -11,27 +11,68 @@ import markdownFragmentToReact from '../utils/markdown-fragment-to-react';
 import { SEO, ThemeProvider } from '../components';
 import markdownComponents from '../markdown-components';
 
-const ReleaseNotesListTemplate = (props) => (
-  <ThemeProvider>
-    <LayoutReleaseNotesList
-      pageContext={props.pageContext}
-      pageData={props.data.contentPage}
-    >
-      <Markdown.TypographyPage>
-        <SEO
-          title={props.data.contentPage.title}
-          excludeFromSearchIndex={props.data.contentPage.excludeFromSearchIndex}
-        />
-        <MDXProvider components={markdownComponents}>
+const ReleaseNotesListTemplate = (props) => {
+  const [releaseNotes, setReleaseNotes] = React.useState(
+    props.data.allReleaseNotePage ? props.data.allReleaseNotePage.nodes : []
+  );
+  const [fromFilterDate, setFromFilterDate] = React.useState('');
+  const [toFilterDate, setToFilterDate] = React.useState('');
+  const [filterTopics, setFilterTopics] = React.useState([]);
+
+  React.useEffect(() => {
+    setReleaseNotes(
+      props.data.allReleaseNotePage.nodes.filter((releaseNote) => {
+        let foundTopicInFilter = true;
+        // console.log('releaseNote.date', releaseNote.date);
+
+        // // todo: use moments to filter out any earlier date
+        // console.log('fromFilterDate', fromFilterDate);
+
+        // // todo: use moments to filter out any later date
+        // console.log('toFilterDate', toFilterDate);
+
+        // console.log(releaseNote.topics);
+
+        if (filterTopics.length > 0) {
+          foundTopicInFilter = releaseNote.topics.find((topic) =>
+            filterTopics.includes(topic)
+          );
+        }
+
+        return foundTopicInFilter;
+      })
+    );
+  }, [
+    props.data.allReleaseNotePage.nodes,
+    fromFilterDate,
+    toFilterDate,
+    filterTopics,
+  ]);
+
+  return (
+    <ThemeProvider>
+      <LayoutReleaseNotesList
+        pageContext={props.pageContext}
+        pageData={props.data.contentPage}
+        handleOnFromFilterDateChange={setFromFilterDate}
+        handleOnToFilterDateChange={setToFilterDate}
+        handleOnFilterTopicsChange={setFilterTopics}
```

Comment:
"handle" is usually the function handling the callback. Here you want to pass an "on"-something prop. For instance, it's the same as `onClick`, `onChange`, etc.

```suggestion
        onFromFilterDateChange={setFromFilterDate}
        onToFilterDateChange={setToFilterDate}
        onFilterTopicsChange={setFilterTopics}
```